packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -89,36 +150,70 @@ export class FungibleTokenTransferController implements ReactiveController {
         }
       }
     });
+
+    this.substrateProviderContext = new ContextConsumer(host, {
+      context: substrateProviderContext,
+      subscribe: true
+    });
+  }
+  get sourceDomainConfig(): EthereumConfig | SubstrateConfig | undefined {
+    if (this.config.environment && this.sourceNetwork) {
+      return this.config.getDomainConfig(this.sourceNetwork.id);
+    }
+    return undefined;
   }
 
   hostDisconnected(): void {
     this.reset();
   }
 
+  /**
+   * Infinite Try/catch wrapper around
+   * {@link Config} from `@buildwithsygma/sygma-sdk-core`
+   * and emits a {@link SdkInitializedEvent}
+   * @param {number} time to wait before retrying request in ms
+   * @returns {void}
+   */
+  async retryInitSdk(retryMs = 100): Promise<void> {
+    try {
+      await this.config.init(1, this.env);
+      this.host.dispatchEvent(
+        new SdkInitializedEvent({ hasInitialized: true })
+      );
+    } catch (error) {
+      setTimeout(() => {
+        this.retryInitSdk(retryMs * 2).catch(console.error);
+      }, retryMs);
+    }
+  }
+
   async init(env: Environment): Promise<void> {
     this.host.requestUpdate();
     this.env = env;
     await this.config.init(1, this.env);
```

Comment:
`await this.retryInitSdk();` why is there an init with hardcoded `chainId`

Reply:
solved